### PR TITLE
Add AttributeTargets.Interface to SkipLocalsInitAttribute

### PR DESCRIPTION
**Pull request by YairHalberstadt**
See https://github.com/dotnet/coreclr/pull/20093#issuecomment-532311218

The test was added: http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Emit.UnitTests/Attributes/AttributeTests_WellKnownAttributes.cs,80a639784e161f94

However it looks like the PR was never updated

### CODE REVIEW

**Comment by tannergooding** [comment]
CC. @agocke for sign off from the language side

**Review by stephentoub — APPROVED** [review]

**Comment by stephentoub** [comment]
LGTM once Andy gives his approval.

**Review by agocke — APPROVED** [review]
LGTM

**Pull request merged**
